src/Parser.js

```diff
@@ -140,9 +140,13 @@ export default class Parser {
      * and fetches the one after that as the new look ahead.
      */
     consume() {
-        this.nextToken = this.gullet.get(this.mode === "math");
+        this.nextToken = this.gullet.get(false);
```

Comment:
Yes, I wanted a second opinion on this.  As I wrote above: "we never use `MacroExpander.get` with a `true` argument, which could simplify the code of both `get` and `unget`."  We no longer need any of the space saving/restoring mechanics, so I'll get rid of that.

"It also means that `switchMode` no longer does anything useful." (in the parser)  This one I'm less sure of.  Maybe `switchMode` would be useful in the future, e.g. if we can ever tweak catcodes in other ways? (e.g. verbatim or url modes?)

Reply:
If you think `switchMode` will come in handy in the future let's keep it, but please add a comment as to why we're keeping it.

Reply:
I think we still need `swtichMode` because we store the current mode in the parse nodes and that value gets used in the builders.  Ignore my comment about adding a comment.